### PR TITLE
Last minute here you go senpai

### PR DESCRIPTION
**Pull request by litehed**
Dancho hasn't finished the drivetrain :(

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonPlsWorkOwO.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        TICKS_PER_REV = fL.getCPR();
+        TICKS_TO_INCHES = Math.PI * WHEEL_DIAMETER / TICKS_PER_REV;
+
+        left = new MotorGroup(fL, bL);
+        right = new MotorGroup(fR, bR);
+
+        diffy = new DifferentialDrive(left, right);
+        diffyOdom = new DifferentialOdometry(() -> fL.getCurrentPosition() * TICKS_TO_INCHES,
+                () -> fR.getCurrentPosition() * TICKS_TO_INCHES,
+                TRACKWIDTH);
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonPlsWorkOwO.java]: Quick way to make this better:
```java
Motor.Encoder leftEncoder = fL.encoder.setDistancePerPulse(TICKS_TO_INCHES);
Motor.Encoder rightEncoder = fR.encoder.setDistancePerPulse(TICKS_TO_INCHES);
rightEncoder.setDirection(Motor.Direction.REVERSE);
diffyOdom = new DifferentialOdometry(leftEncoder::getDistance, rightEncoder::getDistance, TRACKWIDTH);
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonPlsWorkOwO.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        waitForStart();
+        while (opModeIsActive()&& !isStopRequested()) {
+            diffyOdom.updatePosition(5, 5);
+            diffyOdom.updatePose();
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonPlsWorkOwO.java]: So is it just eternally at (5,5)?

Comment (litehed) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonPlsWorkOwO.java]: lol I dont know why I put that there

Comment (JIceberg) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonPlsWorkOwO.java]: Lol bruh

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/DriveSystem.java [hunk 3]
```diff
@@ -17,13 +17,13 @@ public DriveSystem(Motor frontL, Motor frontR, Motor backL, Motor backR){
 
         drive = new MecanumDrive(fL, fR, bL, bR);
     }
-    //not in current use and yuou cant make me get rid of it
+    //not in current use and you cant make me get rid of it
```
[the remark this answers is not in the document]
Comment (litehed) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/DriveSystem.java:20]: the spelling bothered me

**Comment by JIceberg** [comment]
**DISCLAIMER**: So the way you implement it is going to suck because you're using differential odometry.

Example for go-to-point in LinearOpMode (not sure why it isn't a command opmode but lol):
```java
private PIDController xCont, yCont, hCont;

/* instantiation of pid controllers not included */

xCont.setSetPoint(5);
hCont.setSetPoint(Math.PI);
do {
   if (isStopRequested()) break;
   diffy.arcadeDrive(
      xCont.calculate(diffyOdom.getPose().getX()),
      hCont.calculate(diffyOdom.getPose().getHeading())
   );
   diffyOdom.updatePose();
} while (opModeIsActive() && (!xCont.atSetPoint() || !hCont.atSetPoint()));

yCont.setSetPoint(5);
do {
   if (isStopRequested()) break;
   diffy.arcadeDrive(yCont.calculate(diffyOdom.getPose().getY()), 0);
   diffyOdom.updatePose();
} while (opModeIsActive() && !yCont.atSetPoint());
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java [hunk 4]
```diff
@@ -11,16 +11,25 @@
 import java.util.function.DoubleSupplier;
 
 public class ShooterSystem extends SubsystemBase {
-    private MotorEx shooterMotor;
+    private Motor shooterMotor;
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java]: Change this back to MotorEx.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java [hunk 5]
```diff
@@ -11,16 +11,25 @@
...
+    public boolean active() {
+        return shooterActive;
     }
 
+    public void toggle() {
+        shooterActive = !shooterActive;
+    }
     public void shoot(){
         telemetry.addData("Shooter speed", power.getAsDouble());
         telemetry.update();
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java]: Remove this and put it in a periodic method:
```java
@Override
public void periodic() {
    telemetry.addData("Shooter power", power.getAsDouble());
    telemetry.addData("Shooter active", shooterActive);
    telemetry.update();
}
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java [hunk 6]
```diff
@@ -11,16 +11,25 @@
...
 
 public class ShooterSystem extends SubsystemBase {
-    private MotorEx shooterMotor;
+    private Motor shooterMotor;
     private Telemetry telemetry;
     private DoubleSupplier power;
+    private boolean shooterActive;
 
-    public ShooterSystem(MotorEx ShooterMotor, Telemetry telemetryIn, DoubleSupplier getPower){
+    public ShooterSystem(Motor ShooterMotor, Telemetry telemetryIn, DoubleSupplier getPower){
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java]: Change parameter to MotorEx

**Review by JIceberg — COMMENTED** [review]
Where is your opmode that just uses the shooter subsystem that you can use for testing tomorrow?

**Review by JIceberg — APPROVED** [review]

**Pull request merged**